### PR TITLE
Convert `FileBrowser` to TS/Composition API

### PR DESCRIPTION
**Pull request by mvandenburgh**
Also merges the `PublishFileBrowser` component into `FileBrowser.vue` as part of #903.

I also changed the CI config so tests will run on any branch, not just `master` or branches that have PRs associated with them; GitHub Actions minutes are unlimited for public repositories so I don't see why not.

### CODE REVIEW

File: src/rest/publish.ts [hunk 1]
```diff
@@ -99,7 +99,8 @@ const publishRest = new Vue({
         throw error;
       }
     },
-    async assetPaths(identifier: string, version: string, location: string): Promise<string[]> {
+    // TODO: define return type here
+    async assetPaths(identifier: string, version: string, location: string): Promise<any> {
```
Comment (dchiquito) [inline src/rest/publish.ts:103]:
```suggestion
    async assetPaths(identifier: string, version: string, location: string): Promise<{ folders: AssetFolder[], files: AssetFile[] }> {
```
I'm not sure it's worth it's own type, so using a custom type definition seems like the easy way out.
You probably need to import those types in this file though.

File: src/types/index.ts [hunk 2]
```diff
@@ -63,3 +63,26 @@ export interface DandisetStats {
   asset_count: number,
   size: number,
 }
+
+export interface AssetStats {
+  folder: boolean,
+  name: string,
+  created: string|undefined,
+  modified: string|undefined,
+  size: number|undefined,
```
Comment (dchiquito) [inline src/types/index.ts:72]:
```suggestion
  created: string?,
  modified: string?,
  size: number?,
```
This is the convenient syntax for marking something as optional, which is the same as `whatever|undefined`.

File: src/views/FileBrowserView/FileBrowser.vue [hunk 3]
```diff
@@ -204,15 +204,21 @@
...
 import { publishRest } from '@/rest';
 import store from '@/store';
+import { draftVersion } from '@/utils/constants';
+import { AssetStats } from '@/types';
 
 const parentDirectory = '..';
 const rootDirectory = '';
 
-const sortByName = (a, b) => {
+const sortByName = (a: any, b: any) => {
```
Comment (dchiquito) [inline src/views/FileBrowserView/FileBrowser.vue:221]: Aren't these `AssetStats`?

File: src/views/FileBrowserView/FileBrowser.vue [hunk 4]
```diff
@@ -30,30 +256,123 @@ export default {
...
+
+    const currentDandiset = computed(() => store.state.dandiset.publishDandiset);
+    const splitLocation = computed(() => location.value.split('/'));
+
+    const items: Ref<AssetStats[]|null> = ref(null);
+
+    async function getAssets() {
+      const { version, identifier } = props;
+
+      const data = await publishRest.assetPaths(identifier, version, location.value);
```
Comment (dchiquito) [inline src/views/FileBrowserView/FileBrowser.vue:273]:
```suggestion
      const { files, folders } = await publishRest.assetPaths(identifier, version, location.value);
```

File: src/views/FileBrowserView/FileBrowser.vue [hunk 5]
```diff
@@ -30,30 +256,123 @@ export default {
...
     }
-  },
-  methods: {
-    fetchPublishDandiset() {
+
+    watchEffect(async () => {
+      updating.value = true;
+      await getAssets();
+      updating.value = false;
+    }, { flush: 'sync' });
```
Comment (dchiquito) [inline src/views/FileBrowserView/FileBrowser.vue:305]: What does `flush: 'sync'` mean? Never seen that before

**Review by dchiquito — CHANGES_REQUESTED** [review]
Some nitpicks

File: src/types/index.ts (hunk 2 above)
Comment (mvandenburgh) [reply src/types/index.ts]: https://github.com/dandi/dandiarchive/pull/904/commits/77269760e88c29e960c08158ba518fdb104b711d

**Comment by mvandenburgh** [comment]
@dchiquito this should be all set for you to review again, PTAL. FYI to save you some time, the only code that changed since your last review is the `assetPaths()` return type in `rest.ts` and the contents of the `<script>` section of `FileBrowser.vue`. Mentioning this b/c I removed a redundant top-level `<div>` in that component and now the diff view is showing the entire template as new code.

**Review by dchiquito — APPROVED** [review]

**Pull request merged**